master/internal/api_experiment.go

```diff
@@ -2100,11 +2100,16 @@ func sortExperiments(sortString *string, experimentQuery *bun.SelectQuery) error
 				"."+metricQualifier)
 			if !slices.Contains(SummaryMetricStatistics, metricQualifier) {
 				return status.Errorf(codes.InvalidArgument,
-					"sort training metrics by statistic: count, last, max, min, or sum")
+					"sort training metrics by statistic: last, max, min, or mean")
+			}
+			if metricQualifier == "mean" { //nolint: goconst
+				locator := bun.Safe(fmt.Sprintf("trials.summary_metrics->'avg_metrics'->'%s'", metricName))
+				experimentQuery.OrderExpr("(?0->>'sum')::float8 / (?0->>'count')::int ?1",
+					locator, bun.Safe(sortDirection))
+			} else {
+				experimentQuery.OrderExpr("trials.summary_metrics->'avg_metrics'->?->>? ?",
```

Comment:
**question:** it seems like `%s` might be more specific compared to `?` to explicitly limit the param to strings? Or is `?` something unique and required for Bun parameters?

Reply:
sprintf's `%s` is a straightforward string substitution.
`?` in bun will be escaped as a value by default, as a SQL identifier with `bun.Ident`, or not escaped for `bun.Safe`.